pkg/smokescreen/stats_server.go

```diff
@@ -47,6 +48,18 @@ func (s *StatsServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	s.mux.ServeHTTP(w, req)
 }
 
+func (s *StatsServer) GetNumActiveConn() int {
+	n := 0
+	s.config.ConnTracker.Range(func(k, v interface{}) bool {
+		c := k.(*ConnExt)
+		if time.Now().Sub(c.LastActivity) < s.config.IdleThresholdSec {
```

Comment:
The stats lock on the instrumented connection (ConnExt) should be grabbed before accessing fields in it. ⚠️